group/edwards25519/scalar.go

```diff
@@ -4,11 +4,196 @@
 
 package edwards25519
 
+import (
+	"crypto/cipher"
+	"errors"
+	"io"
+	"math/big"
+
+	"gopkg.in/dedis/kyber.v1"
+	"gopkg.in/dedis/kyber.v1/group/mod"
+	"gopkg.in/dedis/kyber.v1/util/bytes"
+	"gopkg.in/dedis/kyber.v1/util/marshalling"
+	"gopkg.in/dedis/kyber.v1/util/random"
+	"gopkg.in/dedis/kyber.v1/util/subtle"
+)
+
 // This code is a port of the public domain, "ref10" implementation of ed25519
 // from SUPERCOP.
 
 // The scalars are GF(2^252 + 27742317777372353535851937790883648493).
 
+type scalar struct {
+	v [32]byte
+}
+
+// Equality test for two Scalars derived from the same Group
+func (s *scalar) Equal(s2 kyber.Scalar) bool {
+	v1 := s.v[:]
+	v2 := s2.(*scalar).v[:]
+	return subtle.ConstantTimeCompare(v1, v2) != 0
+}
+
+// Set equal to another Scalar a
+func (s *scalar) Set(a kyber.Scalar) kyber.Scalar {
+	s.v = a.(*scalar).v
+	return s
+}
+
+func (s *scalar) Clone() kyber.Scalar {
+	s2 := *s
+	return &s2
+}
+
+func (s *scalar) setInt(i *mod.Int) kyber.Scalar {
+	b := i.LittleEndian(32, 32)
+	copy(s.v[:], b)
+	return s
+}
+
+// Set to a small integer value
+func (s *scalar) SetInt64(v int64) kyber.Scalar {
```

Comment:
even if golint doesn't complain - can we still follow go's convention for commenting methods? Please? Pretty Please?